### PR TITLE
Updated automation and golden snapshots after footer change

### PR DESCRIPTION
**Pull request by dogusata**
This PR includes:
- updated test config to allow headful run on pre-push via env variable
- updated golden snapshots for footer text position change
- updated animation end helper to let the rendering finish before the snapshot taken
- updated screenshot comparison threshold for snapshots (increases match strictness)
- updated automation configurations
- added a small delay for the stream end on test app to simulate latency of postMessage
- added e2e tests to publish workflow

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by rahmaniaam — APPROVED** [review]

File: ui-tests/jest.config.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+const isHeadless = process.env.HEADLESS !== 'false';
```
Comment (ege0zcan) [inline ui-tests/jest.config.js:1]: super nit: I think `!!process.env.HEADLESS` does the same thing

File: ui-tests/__test__/helpers.ts [hunk 2]
```diff
@@ -14,7 +14,9 @@ export async function waitForAnimationEnd (page: Page, selector?: string): Promi
             if (elm.getAnimations({ subtree: true })
               .find((animation) => animation.playState === 'running') == null) {
               clearInterval(animationStateCheckInterval);
-              resolve();
+              setTimeout(() => {
```
Comment (ege0zcan) [inline ui-tests/__test__/helpers.ts:17]: nit: can we add comments here explaining the three levels of nested timeouts

File: ui-tests/jest.config.js (hunk 1 above)
Comment (dogusata) [reply ui-tests/jest.config.js:1]: Also for strings?

Comment (dogusata) [reply ui-tests/jest.config.js:1]: image

**Pull request merged**